hail/src/main/scala/is/hail/asm4s/ClassBuilder.scala

```diff
@@ -488,6 +488,10 @@ class MethodBuilder[C](
   val returnTypeInfo: TypeInfo[_],
   val isStatic: Boolean = false
 ) extends WrappedClassBuilder[C] {
+  require(parameterTypeInfo.length + isStatic.toInt <= 255,
+    s"""Invalid method, methods may have at most 255 arguments, found ${parameterTypeInfo.length + isStatic.toInt}
+       |Return Type Info: $returnTypeInfo
+       |Parameter Type Info: ${parameterTypeInfo.mkString}""".stripMargin)
```

Comment:
`mkString(", ")`? no separator is default.

Reply:
TypeInfo's toString, uses jvm field descriptors representation, no separator makes it a JVM method descriptor parameter list.

https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.3.3